histomicstk/features/compute_morphometry_features.py

```diff
@@ -68,78 +83,126 @@ def compute_morphometry_features(im_label, rprops=None):
-        # negative element in inertia_tensor_eigvals, so insert
-        # "max(0, )" call here before invoking np.sqrt().
-        inertia_tensor_eigvals = rprops[i].inertia_tensor_eigvals
-        major_axis_length = 4 * np.sqrt(
-            max(inertia_tensor_eigvals[0], 0))
-        minor_axis_length = 4 * np.sqrt(
-            max(inertia_tensor_eigvals[-1], 0))
+    for i, nprop in enumerate(rprops):
 
-        fdata.at[i, 'Size.MajorAxisLength'] = major_axis_length
-        fdata.at[i, 'Size.MinorAxisLength'] = minor_axis_length
-
-        # compute Perimeter
-        fdata.at[i, 'Size.Perimeter'] = rprops[i].perimeter
+        # assign some features from sklearn as-is
+        for name in mapped_feats:
+            fdata.at[i, name] = nprop[featname_map[name]]
 
         # compute Circularity
-        numerator = 4 * np.pi * rprops[i].area
-        denominator = rprops[i].perimeter**2
-        if denominator:
+        numerator = 4 * np.pi * nprop.area
+        denominator = nprop.perimeter ** 2
+        if denominator > 0:
             fdata.at[i, 'Shape.Circularity'] = numerator / denominator
-        else:
-            fdata.at[i, 'Shape.Circularity'] = 0  # should this be NaN?
-
-        # compute Eccentricity
-        fdata.at[i, 'Shape.Eccentricity'] = rprops[i].eccentricity
 
-        # compute EquivalentDiameter
-        fdata.at[i, 'Shape.EquivalentDiameter'] = rprops[i].equivalent_diameter
-
-        # compute Extent
-        fdata.at[i, 'Shape.Extent'] = rprops[i].extent
+        # compute fractal dimension (boundary complexity)
+        fdata.at[i, 'Shape.FractalDimension'] = _fractal_dimension(nprop.image)
 
         # compute Minor to Major axis ratio
-        if major_axis_length > 0:
+        if nprop.major_axis_length > 0:
             fdata.at[i, 'Shape.MinorMajorAxisRatio'] = \
-                minor_axis_length / major_axis_length
+                nprop.minor_axis_length / nprop.major_axis_length
         else:
             fdata.at[i, 'Shape.MinorMajorAxisRatio'] = 1
 
-        # compute Solidity
-        fdata.at[i, 'Shape.Solidity'] = rprops[i].solidity
+        # Hu moments, raw and possibly also intensity-weighted
+        fdata.loc[i, hu_cols] = nprop.moments_hu
+        if intensity_wtd:
+            fdata.loc[i, wtd_hu_cols] = nprop.weighted_moments_hu
 
     return fdata
+
+
+def _fractal_dimension(Z):
+    """
+    Calculate the fractal dimension of an object (boundary complexity).
+
+    Source: https://gist.github.com/rougier/e5eafc276a4e54f516ed5559df4242c0
+
+    From https://en.wikipedia.org/wiki/Minkowski–Bouligand_dimension ...
+    In fractal geometry, the Minkowski–Bouligand dimension, also known as
+    Minkowski dimension or box-counting dimension, is a way of determining the
+    fractal dimension of a set S in a Euclidean space Rn, or more generally in
+    a metric space (X, d).
+
+    """
+    # Only for 2d binary image
+    assert(len(Z.shape) == 2)
+    Z = Z > 0
```

Comment:
In some of rougier's code, this comparison is a threshold (q.v., https://github.com/rougier/fractal_dimension_analysis/blob/main/fractal_analysis_fxns.py#L60).  Should we be following that code more closely, or is there a reason for the difference?

Reply:
@manthey Oh it's just because in the example, the images they are reading are grayscale, with dark background. I think it's more sensible to assume the input images are either binary or grayscale where background is darker. regionprops nuclear mask is binary so this doesn't make a difference anyway so I got rid of the extra unused parameter.